### DISCUSSION TITLE
Haskell and Dex

### DISCUSSION THREAD

**Discussion by general-rishkin** [google-research/dex-lang#624] [Q&A]
Excuse this naive question. I am new to Haskell and Dex. 
Am I correct in assuming that Dex is not a fully-fledged general purpose programming language?
If this is the case, is it possible to use Dex from Haskell (I will of course have to learn Haskell too) ?

**Comment by oxinabox** [comment]
> Am I correct in assuming that Dex is not a fully-fledged general purpose programming language?

That would be correct.
It is still very alpha.
Though I think there is nothing much that is impossible.
Check the examples.

> If this is the case, is it possible to use Dex from Haskell (I will of course have to learn Haskell too) ?

It is not.
Dex is written in Haskell, so you can extend Dex by writing Haskell.
But you can't call Dex from Haskell.
You can call Dex from Python, though.

**Reply by general-rishkin** [reply]
Excellent. Many thanks.
